runtime/dom-particle-base.js

```diff
@@ -179,4 +179,21 @@ export class DomParticleBase extends Particle {
   boxQuery(box, userid) {
     return box.filter(item => userid === item.getUserID().split('|')[0]);
   }
+
+  // TODO(sjmiles): there's a debounce() here and a _debounce() in xen-state. This
+  // isn't ideal.
+  // This debounce signals to the arc that this particle isn't done yet; the one
+  // in xen-state is a commodity version that doesn't do this.
+  debounce(key, func, delay) {
+    const subkey = `_debounce_${key}`;
+    if (!this._state[subkey]) {
```

Comment:
There is no `_state` in `dom-particle-base`, this module specifically exists to be xen-free. 

We can move this stuff into `dom-particle` (which isn't crazy because that module is also where we introduce blunt change handling) or we can store the debounce semaphores into some other random property.

Reply:
If I move it into dom-particle I can probably override _debounce.